src/complex/Utilities/KUtilities.hpp

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include "complex/Common/Types.hpp"
+#include "complex/complex_export.hpp"
+
+#include <cmath>
+
+namespace complex::KUtilities
+{
+enum COMPLEX_EXPORT DistanceMetric
+{
+  Euclidean,
+  SquaredEuclidean,
+  Manhattan,
+  Cosine,
+  Pearson,
+  SquaredPearson
+};
+
+/**
+ * @brief The DistanceTemplate class contains a templated function getDistance to find the distance, via a variety of
+ * metrics, between two vectors of arbitrary dimensions. The developer should ensure that the pointers passed to
+ * getDistance do indeed contain vectors of the same component dimensions and start at the desired tuples.
+ */
+template <typename leftDataType, typename rightDataType>
+auto GetDistance(const leftDataType& leftVector, usize leftOffset, const rightDataType& rightVector, usize rightOffset, usize compDims, DistanceMetric distMetric)
```

Comment:
```suggestion
float64 GetDistance(const leftDataType& leftVector, usize leftOffset, const rightDataType& rightVector, usize rightOffset, usize compDims, DistanceMetric distMetric)
```

The return type should probably be explicitly defined so that knows the type returned at a glance instead of having to read the entire function. `dist` is defined in the first line, but if operations are performed on the line the function returns, the return value could be implicitly cast to another type and it wouldn't be obvious why the return value is different than expected. This does not happen in this function, but that is not obvious without reading the definition first.